File: ros/ws01/first_ws/src/ctrl_turtle2_pkg/scripts/ctrl_turtle2.py
#! /usr/bin/env python
# coding:utf8

import rospy
from PyQt5.QtWidgets import (
    QApplication,
)
import sys
from MainWindow import MainWindow

if __name__ == '__main__':
    # ros
    nodeName = "python_client_node"  # python_client_node
    rospy.init_node(nodeName)

    # qt
    app = QApplication(sys.argv)
    window = MainWindow()
    window.show()
    sys.exit(app.exec_())

File: 06ros/ws01/first_ws/src/ctrl_turtle2_pkg/src/ctrl_turtle2.cpp
//
// Created by zcb on 2021/4/10.
//
#include <ros/ros.h>
#include <iostream>

#include <QApplication>
#include "MainWindow.h"

using namespace std;

int main(int argc, char **argv) {
    //1 ros
    string nodeName = "cpp_ctrl_turtle_node";
    ros::init(argc, argv, nodeName);
    ros::NodeHandle node;

    //2 qt
    QApplication app(argc, argv);
    MainWindow *window = new MainWindow(&node);
    window->show();

    return app.exec();
}











